src/Compilers/CSharp/Portable/Binder/Binder_InterpolatedString.cs

```diff
@@ -196,8 +196,17 @@ bool tryBindAsHandlerType([NotNullWhen(true)] out BoundInterpolatedString? resul
             }
         }
 
-        private BoundInterpolatedString BindUnconvertedInterpolatedStringToHandlerType(BoundUnconvertedInterpolatedString unconvertedInterpolatedString, NamedTypeSymbol interpolatedStringHandlerType, BindingDiagnosticBag diagnostics, bool isHandlerConversion)
+        private BoundInterpolatedString BindUnconvertedInterpolatedStringToHandlerType(
+            BoundUnconvertedInterpolatedString unconvertedInterpolatedString,
+            NamedTypeSymbol interpolatedStringHandlerType,
+            BindingDiagnosticBag diagnostics,
+            bool isHandlerConversion,
+            ImmutableArray<BoundInterpolatedStringArgumentPlaceholder> additionalConstructorArguments = default,
+            ImmutableArray<RefKind> refKinds = default)
         {
+            Debug.Assert(additionalConstructorArguments.IsDefault
```

Comment:
>additionalConstructorArguments [](http://example.com/codeflow?start=25&length=30)

It feels like it would be much simpler to disallow passing ```default```  for the last two parameters. If it is convenient to allow that at the call sites, after asserting we can reset values using NullToEmpty helper and simplify the logic below by not worrying about ```default``` arrays.

Reply:
I think it's important for the call sites to enforce this invariant, to make sure that any future locations where need to bind these does the "right" thing, but I agree we can do `NullToEmpty` after the invariants are verified.